### ISSUE TITLE
Native UI

### ISSUE THREAD

**Issue by ckeboss** [meirtsvi/WazeWP7#69]
I have been working on the UI, and an wondering where the direction of this project wants to keep this "custom UI", or to make a more native feel. If so, I will continue to work and submit a pull request. If not, I will re-evaluate my ideas.

**Comment by meirtsvi** [comment]
The direction is to make it more Windows Phone native UI. We already did that for Settings pages using the panorama, we added appbar for frequently used functions and there are still couple of screens not implemented in SL. Keep in mind that in order to wire new SL screen to the existing code you will need to wire it to the C code too which is not trivial task. Take a look at the Settings SL screen for more info.
Thanks for working on that!

**Comment by ckeboss** [comment]
Yes, that was my next question. What I have done is just modified some existing screens and replaced some custom styled text fields and buttons with native look/feel buttons/text fields. I was looking to see where I could see an example of how say the settings panorama is linked to bb_waze_code. Any pointers on where to look? Is there an IRC you hang in?

**Comment by meirtsvi** [comment]
ok, so it goes like this:

First, install Cygwin. Choose devel, python and perl.

Syscalls.cs is the actual bridge between the C code and the C#. Check out OnSettingsSaved inside syscalls to learn how to call C from C# (UIWorker.addUIEvent)

Essentially the common way to communicate between C and C#, the C code uses NOPH_ function calls. These calls are implemented inside the syscalls.cs.
In order to add new NOPH_ prototype you will have to:
   a. Add the prototype in bb_waze_code\j2me\syscalls\rimapi\include\rimapi.h.
   b. Find the place in the C code to call this new NOPH function instead of the C code to draw the dialog by itself. I think you should replace the roadmap_login_profile_dialog_show implementation with a call to the new NOPH function.
   c. Open Cygwin window
   d. cd bb_waze_code/j2me
   e. ./run.sh
   f. Open Visual Studio command prompt window
   g. cd <waze_root>/converted
   h. Run compile.cmd
   i. Implement the new NOPH function in Syscalls.cs
   j. Compile Waze from visual studio
   h. Run it

**Comment by ckeboss** [comment]
I am trying to use Cygwin (noob to it), and the script is not running properly, perhaps a config issue on Cygwin. ./run.sh: line 13: syntax error: unexpected end of file

Any idea?

**Comment by meirtsvi** [comment]
Please make sure that when you install git / git extensions the settings are "check-out as-is, check-in as-is"
